truffle/src/com.oracle.truffle.polyglot/src/com/oracle/truffle/polyglot/PolyglotContextImpl.java

```diff
@@ -913,9 +913,17 @@ private PolyglotLanguageContext getLanguageContextImpl(Class<? extends TruffleLa
         return context;
     }
 
-    @Override
-    public boolean initializeLanguage(String languageId) {
-        PolyglotLanguage language = requirePublicLanguage(languageId);
+    void initializeInternalLanguage(String languageId) {
+        PolyglotLanguage language = engine.idToLanguage.get(languageId);
+        if (language == null) {
+            engine.requirePublicLanguage(languageId);
+            assert false;
+            return;
+        }
+        initializeLanguage(language);
```

Comment:
Use this to enter leave without the embedding error handling. Also the language must never be null so that should be an assertion.
```
void initializeInnerContextLanguage(String languageId) {
        PolyglotLanguage language = engine.idToLanguage.get(languageId);
        assert language != null : "language that created inner context not found";
        Object prev = engine.enterIfNeeded(this);
         try {
              return initializeLanguage(language);
         } finally {
           engine.leaveIfNeeded(this, prev)
        }
}
```